test/reduce.cpp

```diff
@@ -2324,3 +2324,41 @@ TEST(Reduce, nanval_issue_3255) {
     }
     ASSERT_SUCCESS(af_release_array(ikeys));
 }
+
+TEST(Reduce, SNIPPET_algorithm_func_sum) {
+    // clang-format off
+    //! [ex_algorithm_func_sum]
+    //
+    // Create a, a 2x3 array
+    array a = iota(dim4(2, 3));           // a = [0, 2, 4,
+                                          //      1, 3, 5]
+
+    // Create b by summing across the first dimension
+    array b = sum(a);        // sum across the first dimension, same as sum(a, 0)
+
+    // Create c by summing across the second dimension
+    array c = sum(a, 1);     // sum across the second dimension
+
+    // Create d by summing across the third dimension
+    array d = sum(a, 2);     // sum across the third dimension
```

Comment:
Not super useful for this shape of an array. It will just return the same value as the input.